jib-core/src/main/java/com/google/cloud/tools/jib/ProjectInfo.java

```diff
@@ -25,7 +27,8 @@
   /** Link to file an issue against the GitHub repository. */
   public static final String GITHUB_NEW_ISSUE_URL = GITHUB_URL + "/issues/new";
 
-  /** The project version. */
+  /** The project version. May be {@code null} if the version cannot be determined. */
+  @Nullable
```

Comment:
Should this be injected into some config by maven/gradle? Or maybe it's another value used by skaffold files?

Reply:
I think I want to refactor `ProjectInfo.VERSION` out of here. It's kind of an implementation detail that this resolved correctly. I'd rather populate it from the maven/gradle plugins with a reasonable default. But that's not the purpose of this PR.https://github.com/GoogleContainerTools/jib/issues/1716